site/hooks/useUpdatingState.js

```diff
@@ -0,0 +1,15 @@
+import { useState } from 'react'
+
+import { useTimeout } from './useTimeout'
+
+// This is a hook that returns a state that auto-updates periodically by calling
+// `getState`. The update period is set through `delay` and is ms.
+export const useUpdatingState = function (getState, delay) {
+  const [state, setState] = useState(getState())
+
+  useTimeout(function () {
+    setState(getState())
+  }, delay)
```

Comment:
if `dependencies` (the third parameter of `useTimeout`) is `undefined` (as it is here), I'm wondering if that wouldn't trigger the effect in every render. That doesn't mean that it will go to an infinite loop, but on every render it will register and unregister the timeout, until the last render finishes, and then the last timeout will fire. Perhaps with the proper dependencies, we can avoid that.


cc @mbonfoster for 2nd pair of eyes 😆

Reply:
Let's see how we can improve this...

Reply:
Improved in b73e76b.